x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mockConfig, mockLogger } from '../__mocks__';
+
+import { createEnterpriseSearchRequestHandler } from './enterprise_search_request_handler';
+
+jest.mock('node-fetch');
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mock;
+const { Response } = jest.requireActual('node-fetch');
+
+const responseMock = {
+  ok: jest.fn(),
+  notFound: jest.fn(),
```

Comment:
Let's remove `notFound` since we're using customError/502 now:

```suggestion
```

In the future (I'll probably address in my follow-up PR), we should forward whatever status method the API endpoint sends us, so we'll likely be using `custom()` in any case, not necessarily `notFound` specifically. Hope that makes sense!